compiler/src/dotty/tools/dotc/ast/MainProxies.scala

```diff
@@ -94,8 +94,17 @@ object MainProxies {
       val body = Try(call, handler :: Nil, EmptyTree)
       val mainArg = ValDef(nme.args, TypeTree(defn.ArrayType.appliedTo(defn.StringType)), EmptyTree)
         .withFlags(Param)
+      def insertTypeSplices = new TreeMap {
```

Comment:
I don't understand what this method is for, could you add a comment here?

Reply:
Added a comment